src/main/java/org/apereo/openlrs/OpenLRSAuthenticationFilter.java

```diff
@@ -111,11 +111,24 @@ private void authenticateOAuth(String authorizationHeader, HttpServletRequest re
 						normalizedParams.put(key, value);
 					}
 				}
-				
+
+				StringBuffer requestURL = request.getRequestURL();
+				String requestScheme = request.getScheme();
+				String originalScheme = request.getHeader("x-forwarded-proto");
+
+				// If original request scheme (maybe through load-balancer) is different, use it instead.
```

Comment:
Should probably check the port as well.  E.g. if ssl is stripped away the port may change from 443 to 80.